pkg/bgpv1/agent/controller.go

```diff
@@ -174,35 +186,41 @@ func configureForClusterPoolIPAM(factory externalversions.SharedInformerFactory,
 // The constructor requires an implementation of BGPRouterManager to be provided.
 // This implementation defines which BGP backend will be used (GoBGP, FRR, Bird, etc...)
 // NOTE: only GoBGP currently implemented.
-//
-// Cancel the provided CTX to stop the Controller.
-func NewController(ctx context.Context, clientset client.Clientset, rtMgr BGPRouterManager, opts ...ControllerOpt) (*Controller, error) {
+func NewController(params ControllerParams) (*Controller, error) {
+	// If the BGP control plane is disabled, just return nil. This way the hive dependency graph is always static
+	// regardless of config. The lifecycle has not been appended so no work will be done.
+	if !params.DaemonConfig.BGPControlPlaneEnabled() {
+		return nil, nil
+	}
+
+	log.Info("Initializing BGP Control Plane")
```

Comment:
While this was in the original code, I don't think this is a useful thing to log. Hive already logs all start hook calls, so that should be enough.